sktime/transformations/series/detrend/_detrend.py

```diff
@@ -197,36 +183,15 @@ def _inverse_transform(self, X, y=None):
         Xt : pd.Series or pd.DataFrame, same type as X
             inverse transformed version of X
         """
-        fh = ForecastingHorizon(X.index, is_relative=False)
-
-        # univariate: X is pd.Series
-        if isinstance(X, pd.Series):
-            # note: the y in the transformer is exogeneous in the forecaster, i.e., X
-            X_pred = self.forecaster_.predict(fh=fh, X=y)
-            if self.model == "additive":
-                return X + X_pred
-            elif self.model == "multiplicative":
-                return X * X_pred
-        # multivariate: X is pd.DataFrame
-        if isinstance(X, pd.DataFrame):
-            X = X.copy()
-            # check if all columns are known
-            X_fit_keys = set(self.forecaster_.keys())
-            X_new_keys = set(X.columns)
-            difference = X_new_keys.difference(X_fit_keys)
-            if len(difference) != 0:
-                raise ValueError(
-                    "X contains columns that have not been "
-                    "seen in fit: " + str(difference)
-                )
-            for colname in X.columns:
-                X_pred = self.forecaster_[colname].predict(fh=fh, X=y)
-                if self.model == "additive":
-                    X[colname] = X[colname] + X_pred
-                elif self.model == "multiplicative":
-                    X[colname] = X[colname] * X_pred
-
-            return X
+        fh = self._get_fh_from_X(X=X)
+        forecaster = self._get_fitted_forecaster(X=X, y=y, fh=fh)
```

Comment:
If `X` is the de-trended data, wouldn't passing this to `_get_fitted_forecaster` return a newly fitted forecaster when `if self.forecaster_.get_tag("requires-fh-in-fit", True)`, resulting in the forecaster being fitted on the de-trended data? Is there an example forecaster you have in mind where this is needed to add the trend back to the data?

Reply:
ah, good catch! Didn't think carefully about this. I suppose fitting on the detrended data does not really make sense here.

Should it then just raise an error, if `self.forecaster_.get_tag("requires-fh-in-fit", True)`?

Reply:
here's I think what should happen in the `inverse_transform`: the data should *not* be updated with the new data, simply the data in `fit` should be remembered. Changed to this.

Reply:
The only downside is that now we'll be carrying around a copy of the data inside the transformer. Perhaps this is not much of an issue because it's just reference rather than a deep copy of the data. My concern was If I wanted to just use something like the `PolynomialTrendForecaster` to do de-trending and re-trending, then I don't actually require the data after `fit`, I certainly wouldn't want the burden of carrying around a copy of the data unnecessarily. But we don't have the memory burden because it's not a deep copy, so hopefully should be fine.

Reply:
well, it will not happen for the `PolynomialTrendForecaster` because its tag is `"requires-fh-in-fit": False`.

The memorization happens *only* in cases where this tag is `True` - a case where the transformer previously would just crash.

In the `False` case, behaviour is precisely as before for univariate forecasters, and the "bugfixed" behaviour for multivariate or hierarchical ones.